pkg/storage/storage.go

```
@@ -446,20 +489,80 @@ func (s *storageModule) createSubvol(size uint64, name string, poolType pkg.Devi
 		return nil, pkg.ErrInvalidDeviceType{DeviceType: poolType}
 	}
 
-	type Candidate struct {
-		Pool      filesystem.Pool
-		Available uint64
+	// Look for candidates in mounted pools first
+	candidates, err := s.checkForCandidates(size, poolType, true)
+	if err != nil {
+		log.Error().Err(err).Msgf("failed to search conditation on unmounted pools")
+	}
+
+	log.Debug().Msgf("Found %d candidates in mounted pools", len(candidates))
+
+	// If no candidates or found in mounted pools, we check the unmounted pools and get the first one that fits
+	if len(candidates) == 0 {
+		log.Debug().Msg("Checking unmounted pools")
+		candidates, err = s.checkForCandidates(size, poolType, false)
+		if err != nil {
+			log.Error().Err(err).Msgf("failed to search conditation on unmounted pools")
```

I think you should return an error here.